runtime/domain.c

```diff
@@ -1272,6 +1266,7 @@ static void domain_terminate()
   // run the domain termination hook
   caml_domain_stop_hook();
   caml_stat_free(domain_state->ephe_info);
+  caml_free_intern_state();
   caml_free_extern_state();
   caml_free_intern_state(domain_state->intern_state);
```

Comment:
`caml_free_intern_state` is called  twice with different arities...  This can't be right!